tests/system/Validation/StrictRules/ValidationTest.php

```diff
@@ -359,6 +359,28 @@ public function testRunGroupWithCustomErrorMessage(): void
         ], $this->validation->getErrors());
     }
 
+    /**
+     * @see https://github.com/codeigniter4/CodeIgniter4/issues/6245
+     */
+    public function testRunWithCustomErrorsAndAsteriskField(): void
+    {
+        $data = [
+            'foo' => [
+                ['bar' => null],
+                ['bar' => null],
+            ],
+        ];
+        $this->validation->setRules(
+            ['foo.*.bar' => ['label' => 'foo bar', 'rules' => 'required']],
+            ['foo.*.bar' => ['required' => 'Required']]
+        );
+        $this->validation->run($data);
+        $this->assertSame([
+            'foo.0.bar' => 'Required',
+            'foo.1.bar' => 'Required',
```

Comment:
Are you really okay with keys like `'foo.0.bar'`?
It seems a bit difficult to use.

Reply:
But as I known, validation service change the asterisk field to like `'foo.0.bar'`.
As Issue #6245 show that.